src/main/kotlin/build/buf/intellij/module/ModuleKey.kt

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.intellij.module
+
+import java.util.Objects
+import java.util.UUID
+
+/**
+ * [ModuleKey] represents the data stored for a dependency in the `buf.lock` file.
+ * This is equivalent to the `bufmodule.ModuleKey` type in the Buf CLI.
+ */
+data class ModuleKey(val moduleFullName: ModuleFullName, val commitID: UUID, val digest: ModuleDigest? = null) {
```

Comment:
Switched from `BufModuleCoordinates` to a more familiar construct to Buf CLI developers.